### PR TITLE
Add summary and confirmation page

### PR DESCRIPTION
**Pull request by techncl**

### CODE REVIEW

File: app/controllers/TransferConfirmationController.scala [hunk 1]
```diff
@@ -0,0 +1,5 @@
+package controllers
+
+class TransferConfirmationController {
```
Comment (TomJKing) [inline app/controllers/TransferConfirmationController.scala:3]: Does this controlled need the security components?

File: app/views/transferConfirmation.scala.html [hunk 2]
```diff
@@ -0,0 +1,29 @@
+@()(implicit messages: Messages)
+@main(Messages("transferConfirmation.header")) {
+    @defining(play.core.PlayVersion.current) { version =>
+        <div class="govuk-grid-row">
+            <div class="govuk-grid-column-two-thirds">
+                <a href="@routes.DashboardController.dashboard()" class="govuk-back-link">Back</a>
```
Comment (TomJKing) [inline app/views/transferConfirmation.scala.html:6]: Is it correct that the back link will take the user back to the Dashboard page from the Transfer Confirmation page? The mock wireframes take the user back to the Transfer summary page: https://u6vda4.axshare.com/reviewandconfirm.html

File: app/views/transferConfirmation.scala.html [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+                    </button>
+                    <!-- Logout -->
+                    <a href="" class="govuk-button govuk-button--secondary" data-module="govuk-button" role="button" draggable="false">
+                        @Messages("transferConfirmation.logoutLink")
+                    </a>
+                </div>
+            </div>
+        </div>
+    }
+}
```
Comment (TomJKing) [inline app/views/transferConfirmation.scala.html:29]: Missing line break

File: app/views/transferConfirmation.scala.html [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+                    <h1 class="govuk-panel__title">
+                        @Messages("transferConfirmation.transferComplete")
+                    </h1>
+                    <div class="govuk-panel__body">
+                        @Messages("transferConfirmation.yourTransferIsComplete")
+                    </div>
+                </div>
+                <div>
+                    <!-- Dashboard -->
+                    <button class="govuk-button" data-module="govuk-button">
```
Comment (TomJKing) [inline app/views/transferConfirmation.scala.html:18]: Should the `button />` be an `<a />` link back to the Dashboard page? I don't think we will be submitting anything from this page.

File: app/views/transferConfirmation.scala.html [hunk 5]
```diff
@@ -0,0 +1,29 @@
...
+        <div class="govuk-grid-row">
+            <div class="govuk-grid-column-two-thirds">
+                <a href="@routes.DashboardController.dashboard()" class="govuk-back-link">Back</a>
+                @progressIndicator(Messages("transferConfirmation.progress"))
+                <div class="govuk-panel govuk-panel--confirmation">
+                    <h1 class="govuk-panel__title">
+                        @Messages("transferConfirmation.transferComplete")
+                    </h1>
+                    <div class="govuk-panel__body">
+                        @Messages("transferConfirmation.yourTransferIsComplete")
```
Comment (TomJKing) [inline app/views/transferConfirmation.scala.html:13]: There is some additional text on the wireframes before the links to the Dashboard / Logout: https://u6vda4.axshare.com/transfercomplete.html

"_What happens next
We’ve received your transfer and will notify you when your records are in our archive.
Please do not delete any records until then._"

File: app/views/transferSummary.scala.html [hunk 6]
```diff
@@ -0,0 +1,92 @@
...
+            <a href="@routes.DashboardController.dashboard()" class="govuk-back-link">Back</a>
+            @progressIndicator(Messages("transferSummary.progress"))
+            <h1 class="govuk-heading-xl">@Messages("transferSummary.title")</h1>
+            <dl class="govuk-summary-list">
+                <div class="govuk-summary-list__row">
+                    <dt class="govuk-summary-list__key">
+                        @Messages("transferSummary.seriesReference")
+                    </dt>
+                    <dd class="govuk-summary-list__value summary-list__right-column">
+
```
Comment (TomJKing) [inline app/views/transferSummary.scala.html:19]: Assume the content for each of the fields is to be added in a separate PR?

File: conf/messages [hunk 7]
```diff
@@ -55,6 +55,24 @@ checkingRecords.ffid=Identifying file formats
 fileChecksResults.header=Record check results
 fileChecksResults.title=Record check results
 fileChecksResults.description=Here are your results
+transferSummary.header=Transfer Summary
+transferSummary.title=Transfer Summary
+transferSummary.seriesReference=Series reference
+transferSummary.consignmentReference=Consignment reference
+transferSummary.transferringBody=Transferring body
+transferSummary.printableReport=Printable Report
```
Comment (TomJKing) [inline conf/messages:63]: Should "report" be lower case for consistency: `Printable report`

File: conf/messages [hunk 8]
```diff
@@ -55,6 +55,24 @@ checkingRecords.ffid=Identifying file formats
...
+transferSummary.download=Download
+transferSummary.filesUploadedForTransfer=Files uploaded for transfer
+transferSummary.openRecords=I confirm that in this version of TDR, all records that I transfer are open
```
Comment (TomJKing) [inline conf/messages:66]: Is this wording correct: "I can confirm **in this version of TDR**, all records that I transfer are open"

**Review by TomJKing — COMMENTED** [review]
Looks good. Added some minor comments/questions. 

Should add some unit tests for the TransferSummaryController

I'm assuming subsequent PRs will add the mutations/display of the transfer summary information.

Has the wording for the pages been confirmed with Kirsten, Sarra and Rooha? It deviates a bit from the wireframe designs.

File: app/controllers/TransferConfirmationController.scala (hunk 1 above)
Comment (techncl) [reply app/controllers/TransferConfirmationController.scala:3]: Yes, but this is just a placeholder controller since I'm not building this page (at least for this story), I just have to redirect the transfer summary page to it; I guess I could delete it?

File: app/views/transferConfirmation.scala.html (hunk 2 above)
Comment (techncl) [reply app/views/transferConfirmation.scala.html:6]: I think yes, for now as every page directs to the dashboard page. I think we're still waiting on the decision as to where the back button should take you.

File: app/views/transferConfirmation.scala.html (hunk 3 above)
Comment (techncl) [reply app/views/transferConfirmation.scala.html:29]: Thanks!

File: conf/messages (hunk 8 above)
Comment (techncl) [reply conf/messages:66]: Yes, this is the text from the updated [wireframe](https://h81xos.axshare.com/reviewandconfirm.html)

File: app/views/transferSummary.scala.html [hunk 9]
```diff
@@ -0,0 +1,92 @@
...
+                    </dd>
+                </div>
+                <div class="govuk-summary-list__row">
+                    <dt class="govuk-summary-list__key">
+                        @Messages("transferSummary.consignmentReference")
```
Comment (suzannehamilton) [inline app/views/transferSummary.scala.html:24]: Could you remove this row from the table for now? We won't be able to fill in the consignment reference for a while, so it's better not to show the row at all so it doesn't confuse users. See the note at the end of the acceptance criteria on [TDR-223](https://national-archives.atlassian.net/browse/TDR-223).

File: app/views/transferSummary.scala.html [hunk 10]
```diff
@@ -0,0 +1,92 @@
...
+                    <dt class="govuk-summary-list__key">
+                        @Messages("transferSummary.transferringBody")
+                    </dt>
+                    <dd class="govuk-summary-list__value summary-list__right-column">
+
+                    </dd>
+                </div>
+                <div class="govuk-summary-list__row">
+                    <dt class="govuk-summary-list__key">
+                        @Messages("transferSummary.printableReport")
```
Comment (suzannehamilton) [inline app/views/transferSummary.scala.html:40]: Could you remove this row too, for the same reason as in [this comment](https://github.com/nationalarchives/tdr-transfer-frontend/pull/226/files#r512674187)?

File: app/views/transferSummary.scala.html (hunk 6 above)
Comment (techncl) [reply app/views/transferSummary.scala.html:19]: Yes, as this information needs to be retrieved from the database; I'm working on this task now.

File: conf/messages (hunk 7 above)
Comment (techncl) [reply conf/messages:63]: I copied the wireframe, but yes, I agree, it probably should.

File: app/views/transferSummary.scala.html (hunk 9 above)
Comment (techncl) [reply app/views/transferSummary.scala.html:24]: Sure, will do. Sorry, I missed this.

File: app/views/transferSummary.scala.html (hunk 10 above)
Comment (techncl) [reply app/views/transferSummary.scala.html:40]: Sure, will do.

File: app/views/transferConfirmation.scala.html (hunk 5 above)
Comment (techncl) [reply app/views/transferConfirmation.scala.html:13]: Yes, I left that out because this page was just supposed to be a placeholder page, I kept adding stuff (since it was easy) and then before you know, I had done a little too much to it. Should I maybe leave as an indicator to the person that picks it up that it is unfinished/not a lot of time was spent developing it?

File: app/views/transferConfirmation.scala.html (hunk 4 above)
Comment (techncl) [reply app/views/transferConfirmation.scala.html:18]: Yes, I tried with `<a>` but the CSS looked wrong, but I'll remove the button so that the person that picks up the task to implement this page can do it properly.

File: app/views/transferConfirmation.scala.html (hunk 2 above)
Comment (TomJKing) [reply app/views/transferConfirmation.scala.html:6]: OK.

File: app/controllers/TransferConfirmationController.scala (hunk 1 above)
Comment (TomJKing) [reply app/controllers/TransferConfirmationController.scala:3]: I would remove it, if its not needed for this task.

File: app/views/transferConfirmation.scala.html (hunk 5 above)
Comment (TomJKing) [reply app/views/transferConfirmation.scala.html:13]: If its just meant to be a placeholder page and there is a separate JIRA task to create this page, then I'd be tempted to add something to page to make this clear: 
```
...
<span class="govuk-caption-xl">Step 9 of 9</span>
<p>Placeholder page</p>
...
```

File: test/controllers/TransferSummaryControllerSpec.scala [hunk 11]
```diff
@@ -0,0 +1,132 @@
...
+      val transferSummaryPage = controller.transferSummary(consignmentId)
+        .apply(FakeRequest(GET, s"/consignment/$consignmentId/transfer-summary").withCSRFToken)
+
+      playStatus(transferSummaryPage) mustBe NOT_FOUND
+      contentAsString(transferSummaryPage) must include("404")
+    }
+
+    "throws an authorisation exception when the user does not have permission to see a consignment's transfer summary" in {
+      val client = new GraphQLConfiguration(app.configuration).getClient[gc.Data, gc.Variables]()
+      val data: client.GraphqlData = client.GraphqlData(Some(gc.Data(None)), List(GraphQLClient.Error("Error", Nil, Nil, Some(Extensions(Some("NOT_AUTHORISED"))))))
```
Comment (TomJKing) [inline test/controllers/TransferSummaryControllerSpec.scala:101]: Probably want to put a line break into line 101 as its along line as it is.

File: test/controllers/TransferSummaryControllerSpec.scala [hunk 12]
```diff
@@ -0,0 +1,132 @@
...
+    wiremockServer.resetAll()
+    wiremockServer.stop()
+  }
+
+  val langs: Langs = new EnglishLang
+
+  "TransferSummaryController GET" should {
+
+    "render the transfer summary page with an authenticated user" in {
+      val consignmentId = UUID.fromString("c2efd3e6-6664-4582-8c28-dcf891f60e68")
```
Comment (TomJKing) [inline test/controllers/TransferSummaryControllerSpec.scala:45]: The same consigmentId is used in multiple tests, so Id be tempted to move this to a global val to avoid repeated code.

File: app/controllers/TransferConfirmationController.scala (hunk 1 above)
Comment (techncl) [reply app/controllers/TransferConfirmationController.scala:3]: Ok

File: app/views/transferConfirmation.scala.html (hunk 5 above)
Comment (techncl) [reply app/views/transferConfirmation.scala.html:13]: Ok, I've replaced it

File: test/controllers/TransferSummaryControllerSpec.scala [hunk 13]
```diff
@@ -0,0 +1,132 @@
...
+      val controller = new TransferSummaryController(getAuthorisedSecurityComponents,
+        new GraphQLConfiguration(app.configuration), getValidKeycloakConfiguration, langs)
+
+      val client = new GraphQLConfiguration(app.configuration).getClient[gc.Data, gc.Variables]()
+      val consignmentResponse: gc.GetConsignment = new gc.GetConsignment(UUID.randomUUID(), UUID.randomUUID())
+      val data: client.GraphqlData = client.GraphqlData(Some(gc.Data(Some(consignmentResponse))), List())
+      val dataString: String = data.asJson.printWith(Printer(dropNullValues = false, ""))
+      wiremockServer.stubFor(post(urlEqualTo("/graphql"))
+        .willReturn(okJson(dataString)))
+
```
[the remark this answers is not in the document]
Comment (techncl) [reply test/controllers/TransferSummaryControllerSpec.scala:55]: Thanks.

File: test/controllers/TransferSummaryControllerSpec.scala [hunk 14]
```diff
@@ -0,0 +1,132 @@
...
+
+      val transferSummaryPage = controller.transferSummary(consignmentId)
+        .apply(FakeRequest(GET, s"/consignment/$consignmentId/transfer-summary").withCSRFToken)
+
```
[the remark this answers is not in the document]
Comment (techncl) [reply test/controllers/TransferSummaryControllerSpec.scala:59]: Thanks.

File: test/controllers/TransferSummaryControllerSpec.scala [hunk 15]
```diff
@@ -0,0 +1,132 @@
...
+      val transferSummarySubmit = controller.transferSummarySubmit(consignmentId)
+        .apply(FakeRequest(POST, s"/consignment/$consignmentId/transfer-summary").withCSRFToken)
+
+      playStatus(transferSummarySubmit) mustBe BAD_REQUEST
+      contentAsString(transferSummarySubmit) must include("govuk-error-message")
+      contentAsString(transferSummarySubmit) must include("error")
+    }
+  }
+
```
[the remark this answers is not in the document]
Comment (techncl) [reply test/controllers/TransferSummaryControllerSpec.scala:131]: Thanks.

File: test/controllers/TransferSummaryControllerSpec.scala (hunk 11 above)
Comment (techncl) [reply test/controllers/TransferSummaryControllerSpec.scala:101]: Ah, yes. I saved it to a variable

File: test/controllers/TransferSummaryControllerSpec.scala (hunk 12 above)
Comment (techncl) [reply test/controllers/TransferSummaryControllerSpec.scala:45]: Good point! Thanks.

**Review by TomJKing — APPROVED** [review]
